python/lsst/daf/butler/core/datastoreCacheManager.py

```diff
@@ -282,6 +282,35 @@ def should_be_cached(self, entity: Union[DatasetRef, DatasetType, StorageClass])
         """
         raise NotImplementedError()
 
+    @abstractmethod
+    def known_to_cache(self, ref: DatasetRef, extension: Optional[str] = None) -> bool:
```

Comment:
I do wonder whether I should put an underscore in front of this name. It's not a method that we should encourage people using. It's only here as a shortcut for the datastore checking.

Reply:
Yes, the assumption is that this is something that only datastores care about. Hopfully the "don't use this" caveat in the description is enough.